### ISSUE TITLE
BUG: Wrong JSON sent to BigQuery when both integer and float fields in schema

### ISSUE THREAD

**Issue by peshitz** [pydata/pandas-gbq#116]
When both a float and integer field exist, integer fields (int64) are serialized as floats, resulting in an error from BigQuery (because in schema the field is defined as Integer).

```
df = pd.DataFrame([[1,1.1],[2,2.2] ],
                   index=['row 1', 'row 2'],
                   columns=['intColumn','floatColumn'])
 
# correct output
df.to_json()

# incorrect output (method used by pandas-gbq)
row = df.iloc[0]
row.to_json()

# correct output
df[['intColumn']].iloc[0].to_json()
```

**Comment by tswast** [comment]
Seems the problem is with using [DataFrame.iterrows()](http://pandas.pydata.org/pandas-docs/version/0.17/generated/pandas.DataFrame.iterrows.html#pandas.DataFrame.iterrows). It changes the dtypes to float64.

```
In [13]: for index, row in df.iterrows():
    ...:     print(row)
    ...:
intColumn      1.0
floatColumn    1.1
Name: row 1, dtype: float64
intColumn      2.0
floatColumn    2.2
Name: row 2, dtype: float64

In [14]: print(df.dtypes)
intColumn        int64
floatColumn    float64
dtype: object
```

**Comment by maxim-lian** [comment]
>  DataFrame.iterrows(). It changes the dtypes to float64.

It returns `pd.Series`, which can only be one dtype

I think the best way to solve this is to do https://github.com/pydata/pandas-gbq/issues/96

**Comment by tswast** [comment]
Confirmed that I can reproduce this with the following test:

```python
    def test_upload_mixed_float_and_int(self):
        """Test that we can upload a dataframe containing an int64 and float64 column.
        See: https://github.com/pydata/pandas-gbq/issues/116
        """
        test_id = "mixed_float_and_int"
        test_size = 2
        df = DataFrame(
            [[1,1.1],[2,2.2]],
            index=['row 1', 'row 2'],
            columns=['intColumn','floatColumn'])

        gbq.to_gbq(
            df, self.destination_table + test_id,
            _get_project_id(),
            private_key=_get_private_key_path(),
            chunksize=10000)

        result_df = gbq.read_gbq("SELECT * FROM {0}".format(
            self.destination_table + test_id),
            project_id=_get_project_id(),
            private_key=_get_private_key_path())

        assert len(result_df) == test_size
```

It gives the error

```
google.api_core.exceptions.BadRequest: 400 Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 1; errors: 1.
```

**Comment by tswast** [comment]
Fix for this release in version 0.3.1.
